modules/network/src/main/java/org/apache/ignite/internal/network/netty/OutboundEncoder.java

```diff
@@ -98,11 +98,17 @@ private NetworkMessageChunkedInput(
             this.serializationService = serializationService;
             this.msg = outObject.networkMessage();
 
-            List<ClassDescriptorMessage> outDescriptors = outObject.descriptors().stream()
-                    .filter(classDescriptorMessage -> !serializationService.isDescriptorSent(classDescriptorMessage.descriptorId()))
-                    .collect(Collectors.toList());
+            List<ClassDescriptorMessage> outDescriptors = null;
```

Comment:
Doesn't concern your changes, this is not the first time we've used a lazy list, it's probably time to create a class for it.

Reply:
The only option that comes to my mind is using some kind of a supplier to introduce lazyness, but using lambdas in such a way seems to be scary for hot code. What do you have in mind?